lib/tasks/demo.rake

```diff
@@ -14,84 +14,144 @@ namespace :demo do
+        puts "\t\tCreated Offer #{offer.id} in 'receiving' state"
+        offer = create_recieved_offer
+        puts "\t\tCreated Offer #{offer.id} in 'received' state"
       end
     end
 
-    def create_package
-      # Create Packages for Goodcity and Items for Stockit
-      puts "Package:\t\tCreating #{count} Packages for Goodcity and Items for Stockit(with_item, with_set_item, received(published and unpublished) & stockit_package"
-
-      count.times do
-        FactoryGirl.create(:package, :with_item, :package_with_locations, :with_inventory_number)
-      end
-      count.times do
-        FactoryGirl.create(:package, :stockit_package, :with_inventory_number , :package_with_locations)
-      end
-      count.times do
-        FactoryGirl.create(:package, :with_set_item, :package_with_locations)
-      end
+    def create_orders
+      puts "Orders:\tCreating #{count} designated and #{count} dispatched order with orders_packages "
       count.times do
-        FactoryGirl.create(:package, :received, :with_inventory_number)
+        create_designated_packages
+        puts "\t\tCreated Order #{Order.last.id} with packages in Designated State"
+        create_dispatched_packages
+        puts "\t\tCreated Order #{Order.last.id} with packages in Dispatched State"
       end
-      count.times do
-        FactoryGirl.create(:package, :received, :with_inventory_number, :published)
+    end
+
+    def create_submitted_offer
+      offer = FactoryGirl.create(:offer, :with_demo_items, :with_messages, created_by: donor)
+      offer.submit
+      offer
+    end
+
+    def create_reviewing_offer
+      offer = create_submitted_offer
+      User.current_user = reviewer
+      offer.start_review
+      offer
+    end
+
+    def create_reviewed_offer
+      offer = create_reviewing_offer
+      offer.finish_review
+      offer
+    end
+
+    def create_scheduled_offer
+      offer = create_reviewed_offer
+      offer.schedule
+      offer
+    end
+
+    def create_inactive_offer
+      offer = create_reviewing_offer
+      offer.mark_inactive
+      offer
+    end
+
+    def create_closed_offer
+      offer = create_reviewed_offer
+      offer.mark_unwanted
+      offer
+    end
+
+    def create_receiving_offer
+      offer = create_reviewed_offer
+      offer.start_receiving
+      offer
+    end
+
+    def create_recieved_offer
+      offer = create_receiving_offer
+      offer = inventory_offer_packages(offer)
+      offer.receive
+      offer
+    end
+
+    def inventory_offer_packages(offer)
+      offer.reload.items.each do |item|
+        item.accept
+        item.packages.each do |package|
+          loc = Location.pluck(:id).sample
+          package.update(inventory_number: InventoryNumber.available_code, allow_web_publish: true, location_id: loc)
+          package.build_or_create_packages_location(loc, 'create')
+          package.mark_received
+        end
       end
+      offer.update(delivered_by: DELIVERED_BY.sample)
```

Comment:
could you remove this from constants and add it as a factory girl sequence instead?